wasm/tests/test.c

```diff
@@ -210,6 +211,34 @@ void test_opa_free()
     test("heap ptr", base == opa_heap_ptr_get());
 }
 
+void test_opa_memoize()
+{
+    opa_memoize_init();
+
+    opa_memoize_insert(100, opa_number_int(1));
+    opa_memoize_insert(200, opa_number_int(2));
+    opa_value *a = opa_memoize_get(100);
```

Comment:
nit: Since we are going through some extra steps (with that new API) to ensure we don't make an allocation on `opa_memoize_get` I'm wondering if its worth checking that in the unit tests. We don't have benchmarks that would catch that kind of regression (yet), so maybe a more explicit check for that behavior is beneficial?

Reply:
Just to close the loop; after an offline discussion we've decided to leave this alone. We'll be adding proper benchmarks soon and any changes to `get()` are unlikely to regress. Also, the new alloc free constructor is not critical to the implementation.